pyproject.toml

```diff
@@ -4,12 +4,12 @@ requires = [
     "setuptools",
     "wheel",
     "Cython>=0.28.5",
-    "numpy==1.13.3; python_version=='3.6' and platform_system!='AIX' and platform_python_implementation == 'CPython'",
-    "numpy==1.14.0; python_version=='3.6' and platform_system!='AIX' and platform_python_implementation != 'CPython'",
-    "numpy==1.14.5; python_version=='3.7' and platform_system!='AIX'",
-    "numpy==1.17.3; python_version>='3.8' and platform_system!='AIX'",
-    "numpy==1.16.0; python_version=='3.6' and platform_system=='AIX'",
-    "numpy==1.16.0; python_version=='3.7' and platform_system=='AIX'",
-    "numpy==1.17.3; python_version>='3.8' and platform_system=='AIX'",
+    # We follow scipy for much of these pinnings
+    # https://github.com/scipy/scipy/blob/master/pyproject.toml
+    "numpy==1.16.5; python_version=='3.6'",
+    "numpy==1.16.5; python_version=='3.7'",
+    "numpy==1.17.3; python_version=='3.8'",
+    # do not pin numpy on future versions of python to avoid incompatible numpy and python versions
+    "numpy; python_version>='3.9'",
```

Comment:
According to @ogrisel, I would suggest the following changes:

```suggestion
    "numpy==1.16.5; python_version=='3.6'",
    "numpy==1.16.5; python_version=='3.7'",
    "numpy==1.17.3; python_version=='3.8'",
    "numpy==1.19.3; python_version>='3.9'",
```

The wheel builder uses the `pyproject.toml` file to know the build dependencies. Because we do not want that our build depends on any recent changes of the NumPy C-API, we must build the wheels with the oldest supported version of `NumPy`. By this way, we ensure that even users with old versions of `NumPy` can install the binary package of `scikit-learn` that is binary compatible, since we have previously ensured it.

Reply:
I restored the original pyproject.toml and just updated this for python 3.9, I sorted the requirements by Python version to make it more clear what was required. I figured it might make more sense to leave it as it was and only drop when Python 3.6 is dropped because there's more machinery that I was not aware of.